Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -168,6 +168,9 @@ theorem pred_zero : pred 0 = 0 :=
 theorem succ_pred_iff_is_succ {o} : succ (pred o) = o ↔ ∃ a, o = succ a :=
   ⟨fun e => ⟨_, e.symm⟩, fun ⟨a, e⟩ => by simp only [e, pred_succ]⟩
 
+theorem succ_pred_iff_mem_range_succ {o} : succ (pred o) = o ↔ o ∈ range succ := by
+  simpa [eq_comm] using succ_pred_iff_is_succ
```

Comment:
What's this new theorem doing here?

Reply:
I've moved my various tweaks to the `Ordinal.pred` API to #21751.